src/services/jsonSchemaService.ts

```diff
@@ -69,41 +74,44 @@ export interface ISchemaHandle {
 	getResolvedSchema(): Thenable<ResolvedSchema>;
 }
 
-
+const BANG = '!';
+const PATH_SEP = '/';
+type AnymatchTesterWrapper = {tester: AnymatchTester, include: boolean};
 class FilePatternAssociation {
 
 	private readonly uris: string[];
-	private readonly patternRegExps: RegExp[];
-	private readonly isInclude: boolean[];
+	private readonly anymatchTesters: AnymatchTesterWrapper[];
 
 	constructor(pattern: string[], uris: string[]) {
-		this.patternRegExps = [];
-		this.isInclude = [];
+		this.anymatchTesters = [];
 		try {
-			for (let p of pattern) {
-				const include = p[0] !== '!';
+			for (let patternString of pattern) {
+				const include = pattern[0] !== BANG;
```

Comment:
this does seem like a bug... was the intention here to check `patternString[0]` ?